### PR TITLE
[Feature] Remove v1 SHARE code

### PR DESCRIPTION
**Pull request by leb2dg**
Resolves https://github.com/CenterForOpenScience/SHARE/issues/367

## Purpose

Remove SHARE v1 code

## Changes

* Removed all SHARE related code except for the registration form
* Use share v2 on search page
* Open new tab when going to SHARE search results

## Side effects

Will need to map these routes with nginx:

```
Rule(
    '/share/',
    'get',
    {},
    OsfWebRenderer('share_search.mako', trust=False)
),
Rule(
    '/share/help/',
    'get',
    {'help': settings.SHARE_API_DOCS_URL},
    OsfWebRenderer('share_api_docs.mako', trust=False)
),
Rule(  # FIXME: Dead route; possible that template never existed; confirm deletion candidate with ErinB
    '/share_dashboard/',
    'get',
    {},
    OsfWebRenderer('share_dashboard.mako', trust=False)
),
Rule(
    '/share/atom/',
    'get',
    search_views.search_share_atom,
    xml_renderer
),

process_rules(app, [

    Rule('/share/search/', ['get', 'post'], search_views.search_share, json_renderer),
    Rule('/share/stats/', 'get', search_views.search_share_stats, json_renderer),
    Rule('/share/providers/', 'get', search_views.search_share_providers, json_renderer),

], prefix='/api/v1')
```

## Ticket

https://github.com/CenterForOpenScience/SHARE/issues/367

### CODE REVIEW

**Comment by leb2dg** [comment]
@chrisseto, what should happen with the OSF search page?

**Comment by chrisseto** [comment]
@leb2dg Point it to SHARE V2's search API, `.../search/_count`. The SHARE url can just be a setting that gets dropped in by mako.

File: website/static/js/search.js [hunk 1]
```diff
@@ -283,7 +283,8 @@ var ViewModel = function(params) {
         self.currentPage(1);
         self.category(alias);
         if (alias.name === 'SHARE') {
-            document.location = '/share/?' + $.param({q: self.query()});
+            var win = window.open(window.contextVars.shareUrl + 'discover?' + $.param({q: self.query()}), '_blank');
```
Comment (leb2dg) [inline website/static/js/search.js]: @chrisseto, this opens SHARE in a new tab; is there a vulnerability that needs to be handled?

Comment (chrisseto) [inline website/static/js/search.js]: Nice job remembering! Looks like it'll be fine as long as we are directing to a domain we control.

As noted [here](https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/) just to be safe add in the following line before the `win.focus()`.

```js
win.opener = null;
```

Comment (leb2dg) [inline website/static/js/search.js]: Will do, thanks!

File: website/templates/search.mako [hunk 2]
```diff
@@ -343,9 +314,11 @@
 </%def>
 
 <%def name="javascript_bottom()">
+    <% import website %>
```
Comment (mfraezz) [inline website/templates/search.mako]: I wouldn't recommend importing all of `website` in a mako rendering context. The `routes.py` entry for this doesn't use a function, but instead just passes an empty dict to this template -- you can put `'shareUrl': settings.SHARE_URL` in there.

**Comment by mfraezz** [comment]
Grepping through the code for about 10 minutes looking for remnants of forgotten code didn't reveal anything SHARE-related that doesn't still need to be there. 🌟 

One issue with how the new SHARE url is being acquired by the front-end, otherwise LGTM.

File: website/routes.py [hunk 3]
```diff
@@ -903,41 +902,20 @@ def make_url_map(app):
         Rule(
             '/search/',
             'get',
-            {},
+            {'shareUrl': settings.SHARE_URL},
```
Comment (mfraezz) [inline website/routes.py]: :+1:

**Pull request merged**